pkg/azuredisk/azuredisk_v2.go

```diff
@@ -286,6 +286,8 @@ func (d *DriverV2) StartControllersAndDieOnExit(ctx context.Context) {
 		os.Exit(1)
 	}
 
+	podToClaimsMap, claimToPodsMap, volumeToClaimMap, claimToVolumeMap, mutexMap := controller.NewSharedMap(), controller.NewSharedMap(), controller.NewSharedMap(), controller.NewSharedMap(), controller.NewSharedMap()
```

Comment:
Instead of creating and passing these maps as separate arguments and then defining the [`azReconciler`](https://github.com/abhisheksinghbaghel/azuredisk-csi-driver/blob/669992ab3721f782b05a71b73e3c22fd4113e252/pkg/controller/common.go#L82) used by helper functions to access them, consider enapsulating them and the methods operating on them in single struct, e.g. `controllerSharedState`.

Reply:
Sounds good. Will do :)